### DISCUSSION TITLE
Gateway Timeout when connecting to forwarded port

### DISCUSSION THREAD

**Discussion by chrisofspades** [community/community#18152] [Codespaces]
I created a Codespace to work on my Hugo-based website. When I run `hugo server`, it automatically forwards port 1313, but when I try to connect to it I get a 504 Gateway Timeout error:

> 504
> Gateway Timeout
> The port forwarding timed out while trying to create a connection.
> 
> [View docs](https://aka.ms/codespaces-troubleshooting)
> [Service status](https://aka.ms/codespaces-status-page)
> Request ID: bd7d5a96e9728db4d961822a59661c59

How can I resolve this?

**Comment by sayhiben** [comment]
Several members of my team are also experiencing this issue as of this morning. GitHub servers intermittently serve us 503 and 504 responses

![image](https://user-images.githubusercontent.com/678890/172922374-260174be-5bad-47aa-8135-6d25d8d31406.png)

**Comment by honekawa-yamazaki** [comment]
Same issue here, it's really terrible!

**Comment by treeder** [comment]
I get this way too much, Codespaces reliability is really bad, hard to depend on it. Getting it on all my codespaces right now.

**Comment by geoffscott** [comment]
I'm getting a 504 Gateway Timeout message right now, even though the status page says all is normal. No changes to code since it last worked. Tried from two different computers. Also tried restarting the Codespace and rebuilding the container.

![Screen Shot 2022-08-20 at 7 43 30 PM](https://user-images.githubusercontent.com/1619862/185769606-8dd3f2bb-2e2b-4bc9-b0f5-1c08aa1fa2f4.png)

**Reply by treeder** [reply]
Ditto. I'm starting to think the Codespaces team doesn't work evenings or weekends as they don't realize anything is wrong until the workday starts. This was a few days ago, bit worse issue, they didn't notice until 6:51am, a solid 8 hours after it started, then another few hours to fix it. https://twitter.com/treeder/status/1559542583394635776

Which is pretty bad for such a critical service (ie: you can't work if you are using it).

**Reply by woochan-hwang** [reply]
Not working for me right now. I think it's about the third time this happened. Previous issues seemed to resolve after restarting codespaces a few times (might have been lucky with the timing based on what others seem to have experienced). No luck at the moment.

**Reply by samudraneel05** [reply]
Same here

**Reply by surilpatel1** [reply]
Same here. Multiple rebuilds, computer uses, different code - 504 issue whenever trying to run http-server or flask. Occurring constantly since Saturday afternoon PST time.

**Comment by alexander-kastil** [comment]
Same for me ... Microsoft wants to push GitHub as DevOps System but reliability is very bad

**Comment by onepayclick** [comment]
I should search on the internet more often. Wasted the weekend thinking my dockerfile had a configuration issue 🤦🏻‍♂️

**Comment by hattima-tim** [comment]
Same here. I am doing a react project on codespace. Today whenever I am running `npm start` I am getting different errors, including `Gateway Timeout`. These are the other errors I am having

![2022-08-21 15 08 48 hattima-tim-shopping-cart-wvqqr54j4q6cv46w-3000 githubpreview dev 205766611ba3](https://user-images.githubusercontent.com/29122213/185794668-2e56105d-808b-436d-93aa-998090188cb2.png)
![2022-08-21 15 11 14 hattima-tim-shopping-cart-wvqqr54j4q6cv46w github dev d994b2f08e1f](https://user-images.githubusercontent.com/29122213/185794677-a96cd6b5-64d8-4814-9ff9-72de0de8dea7.png)
![2022-08-21 19 54 40 hattima-tim-shopping-cart-p9rwqw4gh754x-3000 githubpreview dev 49fe28a5afa4](https://user-images.githubusercontent.com/29122213/185794697-2c8ffccb-a210-4ecc-ae38-d332b35ce7fc.png)

**Comment by geoffscott** [comment]
Not sure if anyone else has done this yet, but I just reported a new outage here: https://support.github.com/contact/bug-report. I referenced this ticket, but it may help if all of us report this as a current outage.

**Comment by Kallin** [comment]
also experiencing this issue just now, creating a new jekyll github pages site. have reported the issue.

**Comment by joshpled** [comment]
Experiencing this is issue as well, reporting with bug-report

**Comment by treeder** [comment]
It took about 21 hours for them to even acknowledge it: https://twitter.com/githubstatus/status/1561412142330515459

Not an acceptable response time for a service like this.

**Comment by araeubig** [comment]
Actually the same for me with 504 and only a wordpress development environment which works extremely stable - but the error must be on GitHub side! The status page gives me "no errors"

**Reply by dawidcyron** [reply]
Same issue here. Seems like Codespaces really ain't as stable as we could wish for it to be yet

**Reply by azizcanyalcin** [reply]
same.

**Reply by stefandeml** [reply]
+1 ... anyone planning to fix this?

**Reply by araeubig** [reply]
i am wondering about the "green github status".......

**Reply by filandra** [reply]
I had the same issue but it's resolved now.

**Reply by joshspicer** [reply]
Apologies for the interruption and thank you for the reports.  Around ~9:30am PT we mitigated an issue specifically with port forwarding in a single geographic region. Please let me know if there's any further issues that we can look into.

**Comment by mityaua** [comment]
Does anyone still have the same problem? (504 status error)
![Screenshot 2023-09-29 095503](https://github.com/community/community/assets/20481332/380e42bf-5eb5-433f-93a2-400bc8ba6c80)

**Comment by ykurniawan** [comment]
Never work for me, tunnel is working fine, I can coding remotely but not with port forwarding. 504 all the time
